src/scenes/NftDetailPage/NftDetailView.tsx

```diff
@@ -87,6 +90,36 @@ export default function NftDetailView({
 
   const { nft, collection } = collectionNft;
 
+  const nextPress = useKeyDown('ArrowRight');
+  const prevPress = useKeyDown('ArrowLeft');
+  const escapePress = useKeyDown('Escape');
+  const backspacePress = useKeyDown('Backspace');
+
+  const handleBackClick = useBackButton({ username });
+  const { replace } = useRouter();
+
+  const navigateToId = useCallback(
+    (nftId: string) => {
+      void replace(`/${username}/${collection.dbid}/${nftId}`);
+    },
+    [username, collection.dbid, replace]
+  );
+
+  useEffect(() => {
+    if (nextPress && nextNftId) {
+      navigateToId(nextNftId);
+    }
+    if (prevPress && prevNftId) {
+      navigateToId(prevNftId);
+    }
+    if (escapePress || backspacePress) {
+      handleBackClick();
+    }
+    // Placing prevNftId and nextNftId in the dependency list would trigger back again, if the user navigated via collection
+    // E.g. nft detail page (back) -> collection page (back) -> gallery. Is there a better solution here?
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [nextPress, prevPress, escapePress, backspacePress, handleBackClick]);
+
```

Comment:
so thinking about this architecture more, i think we should have the hook receive a callback as an argument to trigger, rather than having an effect in the current component handle it.

example:
```js
// current
const nextPress = useKeyDown('ArrowRight');
const doThing = useCallback(() => { ... })
useEffect(() => {
  if (nextPress) doThing()
}, [nextPress, doThing])

// proposed
const doThing = useCallback(() => { ... })
useKeyboardShortcut('ArrowRight', doThing)
```

this would help encapsulate the side-effects into separate hooks, and the component doesn't need to worry about setting up an effect to handle business logic.

a full refactor might look something like this:

```js
const handleNextPress = useCallback(() => {
    if (nextNftId) navigateToId(nextNftId)
}, [nextNftId])

const handlePrevPress = useCallback(() => {
    if (prevNftId) navigateToId(prevNftId)
}, [prevNftId])

const handleBackClick = useBackButton({ username });

useKeyboardShortcut('ArrowRight', handleNextPress)
useKeyboardShortcut('ArrowLeft', handlePrevPress)
useKeyboardShortcut('Escape', handleBackClick)
```

i _believe this should fix your issue with random actions firing unintentionally. you'd need to update the internals of useKeyboardShortcut a bit